src/print.rs

```diff
@@ -1,66 +1,103 @@
+use colored::Colorize;
 use std::fmt;
 
 use crate::geo::location::Point;
 use crate::geo::orientation::Orientation;
+use crate::mission::Outcome;
 use crate::robot::{Command, Robot};
 
-pub fn robots<I>(stream: I)
+pub fn plan<I>(upper_right: Point, stream: I)
 where
-  I: Iterator<Item = (Robot, Vec<Command>)>,
+    I: Iterator<Item = (Robot, Vec<Command>)>,
 {
-  for (robot, commands) in stream {
-    println!("{}", robot);
-    println!(
-      "{}\n",
-      commands
-        .iter()
-        .map(|c| format!("{}", c))
-        .collect::<Vec<String>>()
-        .join("")
-    );
-  }
+    println!("{}", upper_right);
+    for (robot, commands) in stream {
+        println!("{}", robot);
+        println!(
+            "{}\n",
+            commands
+                .iter()
+                .map(|c| format!("{}", c))
+                .collect::<Vec<String>>()
+                .join("")
+        );
+    }
 }
 
-pub fn outcome(result: Result<Robot, Robot>) {
-  match result {
-    Ok(robot) => println!("{}", robot),
-    Err(robot) => println!("{} LOST", robot),
-  }
+pub fn checks<I>(stream: I)
+where
+    I: Iterator<Item = (Outcome, Result<Outcome, String>)>,
+{
+    for (expected, actual) in stream {
+        if let Err(msg) = actual {
+            return eprintln!("{}", msg);
```

Comment:
Guess this is where my `impl Display` suggestion falls over - difficult to use both stdout & stderr with that approach